mockall_derive/src/mock_trait.rs

```diff
@@ -176,3 +174,45 @@ impl MockTrait {
         )
     }
 }
+
+fn write_join_path(
```

Comment:
It would be more idiomatic if this function allocated and returned a new String.

Reply:
Will update. I was trying to avoid string allocations but then again, I couldn't figure out a way to know the required capacity ahead of time anyway, so even the current single `String::new` approach keeps resizing it internally.